src/ScrollViewGesture.tsx

```diff
@@ -285,89 +280,184 @@ const IScrollViewGesture: React.FC<PropsWithChildren<Props>> = (props) => {
+          const fixed = boundary - panOffset.value;
+          const dynamic = panTranslation - fixed;
+          translation.value = boundary + dynamic * 0.5;
+          return;
+        }
+      }
+
+      const translationValue = panOffset.value + panTranslation;
+      translation.value = translationValue;
+    },
+    [isHorizontal, max, panOffset, infinite, overscrollEnabled, translation, validStart, touching],
+  );
+
+  const onGestureFinish = useCallback(
+    (e: GestureStateChangeEvent<PanGestureHandlerEventPayload>) => {
+      'worklet';
+
+      const { velocityX, velocityY, translationX, translationY } = e;
+      scrollEndVelocity.value = isHorizontal.value ? velocityX : velocityY;
+      scrollEndTranslation.value = isHorizontal.value ? translationX : translationY;
+
+      const totalTranslation = scrollEndVelocity.value + scrollEndTranslation.value;
+
+      if (
+        maxScrollDistancePerSwipeIsSet &&
+        Math.abs(totalTranslation) > maxScrollDistancePerSwipe
+      ) {
+        const nextPage =
+          Math.round(
+            (panOffset.value + maxScrollDistancePerSwipe * Math.sign(totalTranslation)) / size,
+          ) * size;
+        translation.value = withSpring(withProcessTranslation(nextPage), onScrollEnd);
+      } else {
+        endWithSpring(onScrollEnd);
       }
+
+      if (!infinite) touching.value = false;
+    },
+    [
+      size,
+      infinite,
+      touching,
+      panOffset,
+      translation,
+      isHorizontal,
+      scrollEndVelocity,
+      scrollEndTranslation,
+      maxScrollDistancePerSwipeIsSet,
+      maxScrollDistancePerSwipe,
+      endWithSpring,
+      withSpring,
+      onScrollEnd,
+    ],
+  );
+
+  const gesture = useMemo(() => {
+    const panGesture = Gesture.Pan()
+      .onBegin(onGestureBegin)
+      .onUpdate(onGestureUpdate)
+      .onEnd(onGestureFinish);
+
+    const {
+      activateAfterLongPress,
+      activeOffsetXEnd,
+      activeOffsetXStart,
+      activeOffsetYEnd,
+      activeOffsetYStart,
+      avgTouches,
+      cancelsTouchesInView,
+      enabled,
+      enableTrackpadTwoFingerGesture,
+      failOffsetXEnd,
+      failOffsetXStart,
+      failOffsetYEnd,
+      failOffsetYStart,
+      hitSlop,
+      manualActivation,
+      maxPointers,
+      minDist,
+      minPointers,
+      minVelocity,
+      minVelocityX,
+      minVelocityY,
+      shouldCancelWhenOutside,
+    } = panGestureHandlerProps ?? {};
+
+    if (typeof activateAfterLongPress === 'number') {
+      panGesture.activateAfterLongPress(activateAfterLongPress);
     }
 
-    const translationValue = panOffset.value + panTranslation;
-    translation.value = translationValue;
-  }, [
-    isHorizontal,
-    max,
-    panOffset,
-    infinite,
-    overscrollEnabled,
-    translation,
-    validStart,
-    touching,
-  ]);
+    if (typeof activeOffsetXStart === 'number' || typeof activeOffsetXEnd === 'number') {
```

Comment:
check type instead of truthiness in case the provided value is falsey and the default value is not